### PR TITLE
Add Acider Runner

### LINKED ISSUE #283: Acider runner
- [x] #284
- [ ] #285
- [ ] #286

### LINKED ISSUE #285: Acider runner prototype

### LINKED ISSUE #286: Acider runner abilities
- [x] #364 
- [x] #365 
- [x] #366 
- [x] #367 
- [x] #372 
- [x] #368 
- [ ] #426

### LINKED ISSUE #426: For the hive

### LINKED ISSUE #4859: Boiler acid gas should do additional armored damage
## Description

5 if the target isn't humanoid, otherwise it's random to be between 15 and 20 burn and checks bio armor.

**Reproduction**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

The one and only acid rouny.

## Why / Balance

parity! Resolves #283, resolves #285, resolves #286, resolves #426, resolves #4859.

## Technical details

Changes were made to the xeno energy system for custom popups and laying down/late infected logic to not give energy.

CorrodibleComp needed a bool added for tagging something as a structure, and made a public function for applying acid to something in the XenoAcidSystem as that was also needed. Also made it work with energy and checking said structure thing. And also for acid applying mods because rouny is fast at aciding.

XenoAcidSlash system applies a comp (supposed to be acid) on hit. pretty simple. 

XenoForTheHiveSystem and it's shared variant do...alot. There's a client side visualizer system for swapping the sprites out and making the animation speed a bit faster. Not sure how noticable the latter is. The sound from starting the bomb also gets louder the closer it is to boom but also kinda hard to tell how much it's scaling. The action event triggers the appearance change and glow and stuff and starts a countdown only the player sees. Once it reaches 0 without them dying (the update func just skips over the dead like in CM) They go boom. Radius depends on store acid, its circular unlike CM's squares because distance functions but shouldn't matter too much I hope. There's an acid range and burn range. Acid range acids barricades with normal level acid and mobs with the acid effect. Damage is based on Range with max damage being right on the rouny and decaying from there. Once a rouny dies this way they either almost immediately respawn from their corpse if there is no hive core or wait 5s to respawn at hive core. If hive core disappears in this time they spawn like burrowed larva for now. I think they're supposed to fail and not give a burrowed larva in that case but that'll have to be implemented later.

Speed values should somewhat match CM, values tested on discord.

TLDR they go boom

Boiler gas doesn't take armor into account except for the 15-20 damage thing, for that I just got the average of the 2 damages.

## Media

https://github.com/user-attachments/assets/07947677-0ad5-49c3-8715-d570c01df1cc

(Note for the Hive's shout has been corrected since this)

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added Acider Runner, who can apply acid to marines on hit, and gains acid reserves which they can spend on boiler-level acid that can be applied quickly or become a living time bomb For The Hive!
- fix: Corrosive acid no longer ignores barricade resistances for damage.
- fix: Fix boiler gas not doing armor ignoring damage, it now does greater damage with only a small portion being effected by armor.

### CODE REVIEW

**Comment by deltanedas** [comment]
finally rouny suiciding to survivors can have a purpose :trollface:

**Comment by muburu** [comment]
maybe it should jitter around violently when it's counting down to explode, i think that conveys "this thing is going to fucking explode" quite well

**Comment by Vermidia** [comment]
Don't think it's needed with the color + anim, green glow (not visible because NV in the video) and sound.

File: Content.Shared/_RMC14/Xenonids/ForTheHive/XenoRespawnComponent.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public bool RespawnAtCorpse = false;
+
+    [DataField]
+    public EntityCoordinates? CorpseLocation;
+
+    [DataField]
+    public EntProtoId Larva = "CMXenoLarva";
+
+	[DataField]
+	public SoundSpecifier CorpseSound = new SoundPathSpecifier("/Audio/_RMC14/Xeno/xeno_newlarva.ogg");
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/ForTheHive/XenoRespawnComponent.cs:27]:
```suggestion
    [DataField]
    public SoundSpecifier CorpseSound = new SoundPathSpecifier("/Audio/_RMC14/Xeno/xeno_newlarva.ogg");
```

File: Resources/Locale/en-US/_RMC14/xeno/xeno-strains.ftl [hunk 2]
```diff
@@ -19,3 +19,11 @@ rmc-xeno-gardener-description = You trade your choice of resin secretions, your
   - Build short-lived walls
   - Speed up the growth of your fruits.
 rmc-xeno-gardener-popup = The glory of gardening: hands in the weeds, head in the dark, heart with resin.
+
+rmc-xeno-acider-name = Acider
+rmc-xeno-acider-description = You trade your leap, bone chips, long-range sight, and a little speed to gain more health, some armor, and a new organ that fills with volatile acid over time.
+  You gain the ability to:
+  - Cause your enemies to be covered in acid on hit.
+  - Gain acid reserves from slashing your enemies, and passively over time, which can be spent on corrosive acid as strong as a boiler's but applied much faster.
+  - Become a living bomb using all your stored up acid. Surivive for 20 seconds and you explode, coating your enemies and their defenses with acid. You'll reincarnate as a larva afterwards.
```
Comment (DrSmugleaf) [inline Resources/Locale/en-US/_RMC14/xeno/xeno-strains.ftl:28]:
```suggestion
  - Become a living bomb using all your stored up acid. Survive for 20 seconds and you explode, coating your enemies and their defenses with acid. You'll reincarnate as a larva afterwards.
```

**Review by DrSmugleaf — DISMISSED** [review]
Thanks for the PR!

**Comment by Rainbeon** [comment]
You used the evil pull values I am going to die
